static_frame/core/frame.py

```diff
@@ -6657,13 +6658,18 @@ def unset_index(self, *,
             consolidate_blocks:
             columns_constructors:
         '''
+        # disallows specifying names with 'drop=True'
+        if drop and names:
+            raise RuntimeError("Cannot specify `names` when `drop=True`, as the index will not be added back as columns.")
+
         def blocks() -> tp.Iterator[TNDArrayAny]:
             # yield index as columns, then remaining blocks currently in Frame
-            if self._index.ndim == 1:
-                yield self._index.values
-            else:
-                # No recache is needed as it's not possible for an index to be GO
-                yield from self._index._blocks._blocks # type: ignore
+            if not drop:
+                if self._index.ndim == 1:
+                    yield self._index.values
+                else:
+                    # No recache is needed as it's not possible for an index to be GO
+                    yield from self._index._blocks._blocks # type: ignore
             for b in self._blocks._blocks:
                 yield b
```

Comment:
Not part of your changes but this is better as:
```suggestion
            yield from self._blocks._blocks
```

Reply:
Yes much cleaner. Will commit suggestion.